### ISSUE TITLE
Use of urlopen()

### ISSUE THREAD

**Issue by Pascal-0x90** [cisagov/findcdn#20]
# Proposal

The use of `urlopen()` can pose security threats and should be replaced with a safer alternative. 

## Motivation

The codebase should more strictly follow Python convention to hopefully improve maintainability.

## Details

I feel this is something we can leave alone in the code. The user cannot define custom URLs. The security vulnerability comes when the user can define the use of `file://` however, we only let the user have access to the `http://` and `https://` protocols. 
https://github.com/cisagov/findcdn/blob/25eb36eed42c398995f2eb6bf55b20f9dd4008ca/src/findcdn/cdnEngine/detectCDN/cdn_check.py#L131

**Comment by DoctorEww** [comment]
I agree there is not an obvious way to exploit the protocol; I don't see this as an issue.

**Comment by S4lt5** [comment]
@Pascal-0x90  @DoctorEww While it might not be exploitable, I had always understood that urllib (not urrllib3 pip module) is not thread safe. Does that pose any issues here with use in the threadpool?
